src/_guides/technical-writing/efficient-tech-writing-process.md

```diff
@@ -0,0 +1,287 @@
+
+Most of these tools will export OpenAPI right over the top of the previous OpenAPI, overriding any previous changes. This is not ideal, and it’s likely these tools will eventually start offering “merge” functionality, but in the meantime how can you add valuable content to OpenAPI contributions without having it vanish on the next build?
+
+## Modifying OpenAPI Descriptions without losing your changes
+
+Regardless of how your OpenAPI is being changed, there are times when you might want to change some things without wanting to (or being able to) change the original. Whether that's the code-first approach exporting OpenAPI, or some engineering teams using API Design First might produce it somewhere the technical writers cannot access, there will be times you need to modify OpenAPI to improve it for various audiences. 
+
+A working group at the OpenAPI Initiative have released a new concept called "Overlays". This is [separate specification](https://github.com/OAI/Overlay-Specification) but compatible with OpenAPI, and while it's still experimental it can be considered stable enough for people to start using.
+
+```yaml
+# overlays.yaml
+overlay: 1.0.0
+info:
+  title: Overlay to customise API for Protect Earth
+  version: 0.0.1
+actions:
+  - target: '$.info.description'
+    description: Provide a better introduction for our end users than this techno babble.
+    update: >-
+      Protect Earth's Tree Tracker API will let you see what we've been planting and restoring all 
+      around the UK, and help support our work by directly funding the trees we plant or the sites 
+      we restore.
+      To get involved [contact us and ask for an access token](https://protect.earth/contact) then
+      [check out the API documentation](https://protect.earth/api).
+```
+
+Using OpenAPI Overlays you can effectively “patch” an OpenAPI description, pointing to parts of the original document with [JSONPath](https://jsonpath.com/), then adding or updating your content in. You can add as many actions to these overlays as you like, or make multiple overlays.
+
+To work with Overlays you’ll need a tool that understands them, and that’s not all OpenAPI tools as the concept is still very new. Regardless of what API documentation tool you are using, you can use the [Bump CLI](https://github.com/bump-sh/cli) to apply these overlays, and this will produce a new user-facing document. 
+
+```
+$ bump overlay openapi.yaml overlays.yaml > openapi.public.yaml
+```
+
+You can run these commands in continuous integration, and whatever you would have done with the original you can now do with the new `openapi.public.yaml` (or whatever you decide to name it). 
+
+Here’s part of a GitHub Action used to deploy overlay-based improved documentation.
+
+```yaml
```

Comment:
I'm sorry I didn't have time to review this PR but I find this example github action workflow a bit odd compared to what we usually recommend.

- First, when using both `diffs` and `deploys` in the same workflow, we [recommend to split those two actions in two different jobs](https://docs.bump.sh/help/continuous-integration/github-actions/#deploy-documentation--diff-on-pull-requests).
- Second, when using `diffs` to comment PRs we [recommend to listen to the `pull_request` event](https://docs.bump.sh/help/continuous-integration/github-actions/#diff-on-pull-requests-only). I'm not sure what the `workflow_dispatch` event shown here is?
- Third, we don't recommend to use the CLI manually to deploy. Instead we [recommend to use our GH action directly](https://docs.bump.sh/help/continuous-integration/github-actions/#deploy-documentation-only).

Could we have a discussion about this to maybe simplify / change this workflow? The article only talks about “deploy[ing] overlay-based improved documentation” maybe we can share a very simple deploy workflow file? Something like:

```yaml
name: Deploy documentation

on:
  push:
    branches:
      - main

jobs:
  deploy-doc:
    name: Deploy API doc on Bump.sh
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Apply overlay to API document
        run: |
          npx bump-cli overlay api/openapi.bundle.yaml api/overlays.yaml > api/openapi.public.yaml
      - name: Deploy API documentation
        uses: bump-sh/github-action@v1
        with:
          doc: partner-api
          token: ${{secrets.BUMP_TOKEN}}
          file: api/openapi.public.yaml
```

Reply:
Please do update the examples to whatever you prefer. I was boding something together quickly for Protect Earth, then simply copied it over, but you'd know what the best recommended approach is.

Reply:
Sure thing: https://github.com/bump-sh/docs/pull/199. Thanks!